server/autoexec.js

```diff
@@ -253,6 +250,40 @@ Object.defineProperty(Array.prototype, 'toLocaleString',
   }
 });
 
+// Add a polyfill to handle replace's second argument being a function.
+(function() {
+  var replace_ = String.prototype.replace;
```

Comment:
At such point as we have the `new` hack working we should use that here instead of a closure.  It will make it easier to understand*/edit the polyfill.

(*) at least if you understand the `new` hack, anyway!